Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingAccelerationStructurePass.cpp

```diff
@@ -107,6 +111,31 @@ namespace AZ
 
                         frameGraph.UseShaderAttachment(desc, RHI::ScopeAttachmentAccess::Write);
                     }
+
+                    RayTracingFeatureProcessor::BlasInstanceMap& blasInstances = rayTracingFeatureProcessor->GetBlasInstances();
+                    for (auto& blasInstance : blasInstances)
+                    {
+                        if (blasInstance.second.m_isSkinnedMesh)
+                        {
+                            for (auto& blasInstanceSubMesh : blasInstance.second.m_subMeshes)
+                            {
+                                for (auto& geometry : blasInstanceSubMesh.m_blas->GetGeometries())
+                                {
+                                    auto* frameAttachment = geometry.m_vertexBuffer.GetBuffer()->GetFrameAttachment();
+                                    if (frameAttachment)
+                                    {
+                                        auto* scopeAttachment = frameAttachment->GetFirstScopeAttachment();
+                                        if (scopeAttachment)
+                                        {
+                                            RHI::BufferScopeAttachmentDescriptor desc = scopeAttachment->GetDescriptor();
+                                            frameGraph.UseAttachment(
+                                                desc, RHI::ScopeAttachmentAccess::Read, RHI::ScopeAttachmentUsage::Shader);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
```

Comment:
This is necessary to get the dependencies right in the frame graph, but I am not 100 % sure if it works in all cases and it is the ideal way to handle the dependencies.

Reply:
Ok, what do you suggest is the better alternative to this?

Reply:
This will cause perf issues. You do not want the framegraph to add barriers/fences for each blas/sumesh/geometry. We had a similar issue with skinning and the way we resolved it was having one big buffer per mesh stream, so one buffer per pos/normal/tan/bitan/blendweight/blendindices for all the meshes which will be read from so no need to call UseAttachment on them as that is the default state of the resource. The output of skinning went into one buffer m_skinnedMeshOutputStream and I think we are calling UseAttachment on it as this buffer is first written into by one scope and then read from by other scopes. So only one UseAttachment call is made in this case. 
If you has a similar issue where you want the gpu to write some data per mesh and the read it later on use one big buffer and offset into it per mesh and then call Useattachment for the scope that is writing to it and then reading from it.

Reply:
Hi, I just tried implementing this change, but I ran into an issue with the pass validation. The `RayTracingAccelerationPass` does not seem to pick up a valid attachment for during pass creation, even though I also declared the slot in pass template. However, when running the editor I can see the slot having a valid attachment from the `SkinningPass`. I attached a screenshot of this happening inside my editor. Would you happen to know what might be going wrong here?

![image](https://github.com/o3de/o3de/assets/143601175/615f5c81-6c00-4ca0-848c-8318e22687f1)

Reply:
For reference: I also took a capture of the implementation with the suggested changes and the `RayTracingAccelerationStructurePass` is still being executed before the `SkinningPass`. There still seems to be a dependency missing. Should I push my current changes to Github for you to take another look?

![image](https://github.com/o3de/o3de/assets/143601175/7874ea4d-a0d3-4f44-869d-6d9e54699812)

Reply:
Please disregard the last two comments. I found a fix for this now